src/wp-includes/option.php

```diff
@@ -1133,6 +1147,95 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ *  Determines the appropriate autoload value for an option based on input.
+ *
+ *  This function checks the provided autoload value and returns a standardized value
+ *  ('on', 'off', 'auto-on', 'auto-off', or 'auto') based on specific conditions.
+ *
+ * If no explicit autoload value is provided, the function will check for certain heuristics around the given option.
+ * It will return `auto-on` to indicate autoloading, `auto-off` to indicate not autoloading, or `auto` if no clear
+ * decision could be made.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @param string    $option   The name of the option.
+ * @param mixed     $value    The value of the option to check its autoload value.
+ * @param bool|null $autoload The autoload value to check.
+ *                                   Accepts 'on'|true to enable or 'off'|false to disable, or
+ *                                   'auto-on', 'auto-off', or 'auto' for internal purposes.
+ *                                   Any other autoload value will be forced to either 'auto-on',
+ *                                   'auto-off', or 'auto'.
+ *                                   'yes' and 'no' are supported for backward compatibility.
+ * @return string Returns the original $autoload value if explicit, or 'auto-on', 'auto-off',
+ *                or 'auto' depending on default heuristics.
+ */
+function wp_determine_option_autoload_value( $option, $value, $autoload ) {
+
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'on' : 'off';
+	}
+
+	switch ( $autoload ) {
+		case 'yes':
+			return 'on';
+		case 'no':
+			return 'off';
+	}
```

Comment:
This is currently missing the support for `on` and `off`. Those values can't be provided by the developer, but it needs to be supported to cover those values coming from the database.
```suggestion
	switch ( $autoload ) {
		case 'on':
		case 'yes':
			return 'on';
		case 'off':
		case 'no':
			return 'off';
	}
```